src/main/java/in/org/projecteka/hiu/dataflow/model/HealthInfoStatus.java

```diff
@@ -1,8 +1,15 @@
 package in.org.projecteka.hiu.dataflow.model;
 
+import java.util.Arrays;
+
 public enum HealthInfoStatus {
     RECEIVED,
     PROCESSING,
     SUCCEEDED,
-    ERRORED
+    ERRORED;
+
+    public static HealthInfoStatus fromString(String value){
+        return Arrays.stream(HealthInfoStatus.values()).anyMatch(healthInfoStatus -> healthInfoStatus.name().equals(value))
```

Comment:
can't we just do valueOf?

Reply:
Because the status can be empty or null if the data part has not been received at that time.